### PR TITLE
Shutdown Kafka connector gracefully

### PR DESCRIPTION
**Pull request by Azquelt**
Only the last three commits are part of this PR. The rest will be removed when #7990 is merged.

### CODE REVIEW

**Comment by Azquelt** [comment]
#build

File: dev/com.ibm.ws.microprofile.reactive.messaging.kafka/src/com/ibm/ws/microprofile/reactive/messaging/kafka/KafkaIncomingConnector.java [hunk 1]
```diff
@@ -77,6 +91,7 @@ private void validate() {
 
         // Create our connector around the kafkaConsumer
         KafkaInput<String, Object> kafkaInput = new KafkaInput<>(this.kafkaAdapterFactory, kafkaConsumer, this.executor, topics, unackedLimit);
+        kafkaInputs.add(kafkaInput);
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.reactive.messaging.kafka/src/com/ibm/ws/microprofile/reactive/messaging/kafka/KafkaIncomingConnector.java]: should this be synchronized?

File: dev/com.ibm.ws.microprofile.reactive.messaging.kafka/src/com/ibm/ws/microprofile/reactive/messaging/kafka/KafkaOutgoingConnector.java [hunk 2]
```diff
@@ -43,8 +50,18 @@
 
         KafkaProducer<String, Object> kafkaProducer = this.kafkaAdapterFactory.newKafkaProducer(producerConfig);
         KafkaOutput<String, Object> kafkaOutput = new KafkaOutput<>(config.getValue("topic", String.class), kafkaProducer);
+        this.kafkaOutputs.add(kafkaOutput);
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.reactive.messaging.kafka/src/com/ibm/ws/microprofile/reactive/messaging/kafka/KafkaOutgoingConnector.java]: should this be synchronized?

**Comment by Azquelt** [comment]
#build

**Comment by Azquelt** [comment]
Rebased on integration

**Pull request merged**